buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -81,6 +120,8 @@
     private static final String IN_VM_SYSPROP = "tests.inVM";
     private static final String DISTRIBUTION_SYSPROP = "tests.distribution";
 
+    private static final boolean dockerTestEnabled = shouldRunDockerTests();
```

Comment:
Can we try and defer this until we actually need it? Executing `shouldRunDockerTests()` hits the filesystem and if possible we want to avoid any and all configuration time I/O from statically initializing these kinds of things.